UserDetailsClient/UserDetailsClient.Core/MainPage.xaml.cs

```diff
@@ -71,12 +54,19 @@ public MainPage()
 
         private IUser GetUserByPolicy(IEnumerable<IUser> users, string policy)
         {
-            foreach (var user in users)
+            if (users != null)
             {
-                string userIdentifier = Base64UrlDecode(user.Identifier.Split('.')[0]);
-                if (userIdentifier.EndsWith(policy.ToLower())) return user;
+                IUser[] userArray = users.ToArray();
+                IUser currentUser = null;
+                for (int i = 0; i < userArray.Length; i++)
+                {
+                    string identifier = userArray[i].Identifier;
+                    string userIdentifier = identifier.Split('.')[0];
+                    if (userIdentifier.EndsWith(policy.ToLower()))
+                    { currentUser = userArray[i]; }
+                }
+                return currentUser;
```

Comment:
After successful login, the API call would throw and error and require interactive login again. The `Base64UrlDecode` was returning a decoded string, so it was impossible to check for the policy on the end of the `userIdentifier`, and was always returning null.